### ISSUE TITLE
Side pane keyboard shortcut inactive on program launch

### ISSUE THREAD

**Issue by afhpayne** [zim-desktop-wiki/zim-desktop-wiki#923]
Platform - Linux 4.19.72
Version - Zim 0.72.0
Description - If Zim preference "Use <Primary><Space> to switch to the side pane" is enabled, 
on launching the program this functionality is absent.  It is necessary to first mouse click the side pane.  Once done, <primary><space> produces expected results -- i.e. toggling focus between pane and document.

**Comment by introt** [comment]
Can you reproduce this with 0.74?

**Comment by afhpayne** [comment]
Yes, can reproduce on 74.3

**Comment by introt** [comment]
Thanks!

TL;DR: `PageIndexWidget` needs to be "realized and mapped", fixing requires familiarity with Gtk (which I don't have).

---

Cursory print debugging reveals that `toggle_sidepane_focus` correctly calls `focus_sidepane` (shown below), which does get a list from `get_visible_panes`...

https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/11d78a509307191457c77b7a7ea3afbb4af9df86/zim/gui/widgets.py#L2735-L2740

...the first item of which is a `widgets.WindowSidePane` object, whose `grab_focus` (shown below) gets called, and the execution proceeds to line 2200, where the `widget` whose `grab_focus` is called is a `pageindex.PageIndexWidget` object.

https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/11d78a509307191457c77b7a7ea3afbb4af9df86/zim/gui/widgets.py#L2194-L2204

The `PageIndexWidget` inherits`grab_focus` from `Gtk.Widget` (through `Gtk.VBox`). [Its documentation](https://lazka.github.io/pgi-docs/Gtk-3.0/classes/Widget.html#Gtk.Widget.grab_focus) gives us the following insight:

![image](https://user-images.githubusercontent.com/26069787/153561026-bd6c01d3-a361-47da-b601-e90c6569aca7.png)

The widget's `get_can_focus` returns `False`; it can be forced to `True` by adding `widget.set_can_focus(True)`, but this results in weird behavior where the focus only kinda moves to the side pane, allowing you to switch between eg. the Index and the Journal as well as selecting dates from the calendar, but not letting you interact with the `GtkTreeView` containing the pages.

**Comment by afhpayne** [comment]
The word "index" could become a focusable widget and its behavior could be to open the panel. Since the panel is open anyway, activating it wouldn't do anything. I would act as a dummy Gtk.Entry so self would have something to point to.
